scripts/spell-check.sh

```diff
@@ -0,0 +1,76 @@
+#!/bin/sh
+#
+## Copyright (C) 2020 The Squid Software Foundation and contributors
+##
+## Squid software is distributed under GPLv2+ license and includes
+## contributions from numerous individuals and organizations.
+## Please see the COPYING and CONTRIBUTORS files for details.
+##
+
+#
+# This script runs codespell against selected files.
+#
+# Use -q or --quiet option to run quietly
+#
+
+CODESPELL_LOC=`which codespell`
+if test "${CODESPELL_LOC}" = ""; then
+    echo "This script requires codespell which was not found."
+    exit 1
+fi
+
+CODESPELL_VER=`codespell --version 2>&1`
+echo "The codespell version is ${CODESPELL_VER}."
+
+UNSTAGED_CHANGES=`git diff | wc -l`
+if test "${UNSTAGED_CHANGES}" != "0"; then
```

Comment:
Does the suggestion below work? If we can use a simpler, more reliable interface, we should. I am also concerned that comparing raw `wc` output with a literal string may not always work as intended.

```suggestion
if ! git diff --quiet; then
```

Reply:
BTW, the current code would proceed when run in a directory _not_ controlled by git. That case is common for scripts executed in sources unpacked from a tarball. Proceeding in such a case is relatively safe because `git ls-files` will also fail later.

One could argue that it is better to stop sooner because, for example, future script changes may add support for applying codespell to files or directories supplied on the command line. The suggested `! git diff --quiet` check has that "exit ASAP" effect. One could argue also that the echoed `There are unstaged changes` message is a bit odd in that case, but it is possible to view all files in such a directory as effectively unstaged (and git's own `Not a git repository` message would help clarify things).

Reply:
Applied in next commit.